lib/LanguageServerCodeTransform/CodeAction/GenerateMethodProvider.php

```diff
@@ -0,0 +1,168 @@
+                array_filter($diagnostics, function (Diagnostic $diag) use ($range) {
+                    return (
+                        $diag->range->start->line <= $range->start->line &&
+                        $diag->range->start->character <= $range->start->character &&
+                        $diag->range->end->line >= $range->end->line &&
+                        $diag->range->end->character >= $range->end->character
+                    );
+                })
+            );
+
+            if (count($diagnostics) === 0) {
+                return [];
+            }
+
+            return [
+                CodeAction::fromArray([
+                    'title' =>  'Generate method (if not exists)',
+                    'kind' => self::KIND,
+                    'diagnostics' => $diagnostics,
+                    'command' => new Command(
+                        'Generate method',
+                        GenerateMethodCommand::NAME,
+                        [
+                            $textDocument->uri,
+                            PositionConverter::positionToByteOffset($range->start, $textDocument->text)->toInt()
+                        ]
+                    )
+                ])
+            ];
+        });
+    }
+    /**
+     * @return array<Diagnostic>
+     */
+    private function getDiagnostics(TextDocumentItem $textDocument): array
+    {
+        $node = $this->parser->parseSourceFile($textDocument->text);
+        $diagnostics = [];
+        foreach ($node->getDescendantNodes() as $node) {
+            if ((!$node instanceof CallExpression)) {
+                continue;
+            }
+            assert($node instanceof CallExpression);
+
+            $memberName = null;
+            if ($node->callableExpression instanceof MemberAccessExpression) {
+                $memberName = $node->callableExpression->memberName;
+            } elseif ($node->callableExpression instanceof ScopedPropertyAccessExpression) {
+                $memberName = $node->callableExpression->memberName;
+            }
+            
+            if (!($memberName instanceof Token)) {
+                continue;
+            }
+
+            assert($memberName instanceof Token);
+            
+            $diagnostics[] = new Diagnostic(
+                new Range(
+                    PositionConverter::byteOffsetToPosition(
+                        ByteOffset::fromInt($memberName->start),
+                        $textDocument->text
+                    ),
+                    PositionConverter::byteOffsetToPosition(
+                        ByteOffset::fromInt($memberName->start + $memberName->length),
+                        $textDocument->text
+                    ),
+                ),
+                'Generate method',
+                DiagnosticSeverity::INFORMATION,
+                null,
+                'phpactor'
+            );
+        }
+
+        usort($diagnostics, function (Diagnostic $a, Diagnostic $b) {
```

Comment:
If not sorted it becomes untestable because the order that `TolerantParser` returns the method calls is sometimes different from the order they occur in the source (purely as line/char).